### ISSUE TITLE
Can I use this project inside FlatList's item?

### ISSUE THREAD

**Issue by YKK101** [fram-x/FluidTransitions#8]
I write ItemList component and when select item, navigator will push to Detail component and show selected image in full size
this is my code

ItemList
```
class ItemList extends Component {
  renderItem = ({ item, index }) => {
    return (
      <TouchableOpacity
        style={styles.cell}
        onPress={() => { this.props.navigation.navigate('DETAIL', { item, index }) }}
      >
          <Transition shared={`image_${index}`}>
            <Image
              style={styles.image}
              source={{ uri: item.uri }}
            />
          </Transition>
      </TouchableOpacity>
    )
  }

  render() {
    return (
      <View style={styles.container}>
        <FlatList
          data={itemData}
          renderItem={this.renderItem}
          keyExtractor={(_, index) => `${index}`}
          numColumns={3}
        />
      </View>
    )
  }
}
```

Detail
```
const Detail = (props) => (
  <View style={styles.container} >
    <Transition shared={`image_${props.navigation.state.params.index}`}>
      <Image
        style={styles.image}
        source={{ uri: props.navigation.state.params.item.uri }}
      />
    </Transition>
  </View>
)
```

![screen shot 2561-04-16 at 10 48 51](https://user-images.githubusercontent.com/12131753/38789338-cb354a62-4163-11e8-81ef-427e8c53a7a7.png)

but when I run my project, the animations works great but my image not appear after I provide it with Transition

![out](https://user-images.githubusercontent.com/12131753/38789276-7998a046-4163-11e8-8055-d377ed300100.gif)

**Comment by chrfalch** [comment]
It's a little bit hard to find out what might go wrong here. Would it be possible for you to provide a reproduction using [Snack](https://snack.expo.io)?

**Comment by dtuan** [comment]
@YKK101 I found that if you try to render Transition view while it is not visible, then the problem happens, not sure this's a good fix:
```
// TransitionView.js
....
const routeDirection = getDirectionForRoute(this._getName(), this._route);
if (routeDirection === RouteDirection.unknown) return { opacity: 1 };  // change opacity to 1
```
hope this help

**Comment by YKK101** [comment]
Right, you can found full version of code [here](https://snack.expo.io/@kanya101/playfluidnavigator?session_id=snack-session-rJHQB-m2f)

**Comment by YKK101** [comment]
@dtuan thanks, did you mean TransitionView.js inside node_modules?
I'll try this and tell you later

**Comment by dtuan** [comment]
@YKK101  yes it's

**Comment by chrfalch** [comment]
I'm currently looking into this issue to see if I can find an easy fix.

**Comment by chrfalch** [comment]
Added fix in PR #10
